search-app/src/cmr/search/services/generic_association_service.clj

```diff
@@ -130,7 +130,12 @@
   associated_item: {concept_id: C6-PROV1}}."
   [mdb-context association operation]
   ;; save each association if there is no errors on it, otherwise returns the errors.
-  (let [association (sort-association association)
+  (let [associated-item-concept-id (:concept-id association)
+        revision-id (:revision-id association)
+        associated-item-revision-id (if (string? revision-id)
+                                      (read-string revision-id)
+                                      revision-id)
+        association (sort-association association)
         {source-concept-id :source-concept-id
```

Comment:
I don't understand what 
        {source-concept-id :source-concept-id
         source-revision-id :source-revision-id
         concept-id :concept-id
         revision-id :revision-id
         errors :errors} association
is doing in the let and how it is being used in the code after it:  Can you please explain it to me?

Reply:
native-id is unique in an association concept, just like any other concepts. it's constructed as <source-id>/<source-revision-id>/<destination-id>/<destination-revision-id>. If we don't "sort", then depending on how the request is constructed, the same two concepts/revisions could end up creating two different association concepts, which is what we try to avoid. Either you associate A with B, or B with A, it should end up being the same association. After we sort the two concepts, they will be inserted into the metadata-db.